File: CSharp/FaculdadeVincit.DesignPatterns.CommandPattern/LuzLigarCommand.cs
namespace FaculdadeVincit.DesignPatterns.CommandPattern;

public class LuzLigarCommand : ICommand
{
    public Luz Luz { get; private set; }

    public LuzLigarCommand(Luz luz)
    {
        Luz = luz;
    }

    public void Executar() => Luz.Ligar();
    public void Desfazer() => Luz.Desligar();

}

File: Java/com.faculdadevincit.designpatterns.commandpattern/src/LuzLigarCommand.java
public class LuzLigarCommand implements ICommand{
    private Luz luz;

    public LuzLigarCommand(Luz luz) {
        this.luz = luz;
    }

    @Override
    public void executar() {
        luz.ligar();
    }

    @Override
    public void desfazer() {
        luz.desligar();
    }

}
